File: Python/program39.py
def Check_Perfect(iNo):
    
    if(iNo < 0):
        iNo = -iNo 
        
    sum = 0
    for iCnt in range(1,(iNo // 2) + 1):
        if((iNo % iCnt) == 0):
            sum = sum + iCnt
    
    if(iNo == sum):
        return True
    else:
        return False
        
           
def main():
    
    iValue = int(input("Enter number\n"))
    
    bRet = Check_Perfect(iValue)
    
    if(bRet == True):
        print(iValue, " is perfect number\n")
    else:
        print(iValue, " is not perfect number\n")

if __name__ == "__main__":
    main()

File: C++/program39.cpp
#include<iostream>
using namespace std;

int CountDigit(int iNo)
{
    int  iCount = 0;

    while(iNo != 0)
    {
        iCount++;
        iNo = iNo / 10;
    }

    return iCount;

}

int main()
{
    int iValue= 0, iRet = 0;

    cout<<"Enter number"<<'\n';
    cin>>iValue;

    iRet = CountDigit(iValue);

    cout<<"Number of digits are "<<iRet<<'\n';

    return 0;
}